src/Twilio/TwilioOrgsTokenAuth.cs

```diff
@@ -0,0 +1,161 @@
+using Twilio.Clients;
+using Twilio.Clients.BearerToken;
+using Twilio.Exceptions;
+using Twilio.Http.BearerToken;
+
+namespace Twilio
+{
+    /// <summary>
+    /// Default Twilio Client for bearer token authentication
+    /// </summary>
+    public class TwilioOrgsTokenAuthClient
+    {
+        private static string _accessToken;
+        private static string _region;
+        private static string _edge;
+        private static TwilioBearerTokenRestClient _restClient;
+        private static string _logLevel;
+        private static TokenManager _tokenManager;
+        private static ClientProperties clientProperties;
+
+        private TwilioOrgsTokenAuthClient() { }
+
+        /// <summary>
+        /// Initialize base client with username and password
+        /// </summary>
+        public static void Init(string grantType, string clientId, string clientSecret)
+        {
+            validateCredentials(grantType, clientId, clientSecret);
+            _tokenManager = new OrgsTokenManager(grantType, clientId, clientSecret);
```

Comment:
Steps.
1. Init 
2. Call Orgs API
3. Init with different client_id and client_secret.
4. call Orgs API
It should pick new client_id and client_secret